cpp/src/orderby/orderby.cu

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018-2019 BlazingDB, Inc.
+ *     Copyright 2018 Jean Pierre Huaroto <jeanpierre@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <type_traits>
+
+#include "cudf.h"
+#include "utilities/cudf_utils.h"
+#include "utilities/error_utils.h"
+
+#include "rmm/thrust_rmm_allocator.h"
+
+#include "../sqls/sqls_rtti_comp.h"
+
+namespace{ //annonymus
+
+  // thrust::device_vector set to use rmmAlloc and rmmFree.
+  template<typename T>
+  using Device_Vector = thrust::device_vector<T, rmm_allocator<T>>;
```

Comment:
Please replace with new rmm::device_vector. No need for a using statement anymore, just use it where you create an array.